fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/highlight/LuceneLocaleTest.java

```diff
@@ -0,0 +1,135 @@
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.highlight;
+
+import com.apple.foundationdb.record.RecordCursor;
+import com.apple.foundationdb.record.ScanProperties;
+import com.apple.foundationdb.record.lucene.LuceneIndexTestUtils;
+import com.apple.foundationdb.record.lucene.synonym.EnglishSynonymMapConfig;
+import com.apple.foundationdb.record.lucene.synonym.SynonymMapRegistryImpl;
+import com.apple.foundationdb.record.metadata.Index;
+import com.apple.foundationdb.record.provider.foundationdb.FDBIndexedRecord;
+import com.apple.foundationdb.record.provider.foundationdb.FDBQueriedRecord;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordContext;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStore;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreTestBase;
+import com.apple.foundationdb.record.provider.foundationdb.IndexOrphanBehavior;
+import com.apple.foundationdb.record.query.plan.QueryPlanner;
+import com.google.protobuf.Message;
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import static com.apple.foundationdb.record.lucene.LuceneIndexTestUtils.TEXT_AND_STORED_COMPLEX;
+import static com.apple.foundationdb.record.provider.foundationdb.indexes.TextIndexTestUtils.COMPLEX_DOC;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Tests to make sure Lucene isn't using the default system locale.
+ * <p>
+ *     At some point, we may want to make the locale used for query parsing into an index option, or
+ *     {@link com.apple.foundationdb.record.lucene.LuceneRecordContextProperties}, but for now it is just set to
+ *     {@link Locale#ROOT}.
+ * </p>
+ */
+public class LuceneLocaleTest extends FDBRecordStoreTestBase {
+
+    @BeforeAll
+    public static void setup() {
+        //set up the English Synonym Map so that we don't spend forever setting it up for every test, because this takes a long time
+        SynonymMapRegistryImpl.instance().getSynonymMap(EnglishSynonymMapConfig.ExpandedEnglishSynonymMapConfig.CONFIG_NAME);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"default", "en_US", "de", "de_DE", "fr"})
+    void highlightedNumberRangeQuery(Locale locale) {
+        final Locale defaultLocale = Locale.getDefault();
+        try {
+            Locale.setDefault(locale);
+            try (FDBRecordContext context = openContext()) {
+                rebuildIndexMetaData(context, COMPLEX_DOC, TEXT_AND_STORED_COMPLEX);
+                recordStore.saveRecord(LuceneIndexTestUtils.createComplexDocument(1623L, "Hello record layer", "Hello record layer 2", 5, 1_234_097, false, 8.123));
```

Comment:
May be interesting to add a record with a number that is outside the search range

Reply:
Yeah, makes sense. I added two records, one where everything is greater, and one where everything is smaller.